juriscraper/state/texas/common.py

```diff
@@ -0,0 +1,681 @@
+        coa_parent = parent.find(
+            './/*[@id="ctl00_ContentPlaceHolder1_COAOnly"]'
+        )
+        children = parent.iterfind('.//*[@class="row-fluid"]')
+
+        if coa_parent is not None:
+            children = chain(
+                children, coa_parent.iterfind('.//*[@class="row-fluid"]')
+            )
+
+        return {
+            self._extract_case_data_name(child.find(".//*[1]")): clean_string(
+                get_all_text(child.find(".//*[2]"))
+            )
+            for child in children
+        }
+
+    def _find_party_in_case_name(self, party_name: str) -> tuple[int, int]:
+        """
+        Finds the start and end indices of the party name in the case name.
+
+        Useful in instances where a name appears as "last, first" in the parties list and "first last" in the case name.
+
+        :param party_name: The party name to search for.
+
+        :return: Index of the party name in the case name.
+        """
+        party_name = harmonize(party_name).lower()
+        if party_name[:3] == "the":
+            party_name = party_name[3:]
+        party_name_parts = [part.strip() for part in party_name.split(",")]
+        if len(party_name_parts) == 1:
+            start = self.case_name_full.lower().find(party_name_parts[0])
+            end = start + len(party_name_parts[0])
+            return start, end
+        party_name_parts_indexed = [
+            (self.case_name_full.find(part), part) for part in party_name_parts
+        ]
+        party_name_parts_indexed.sort(key=lambda x: x[0])
+        start = party_name_parts_indexed[0][0]
+        end = party_name_parts_indexed[-1][0] + len(
+            party_name_parts_indexed[-1][1]
+        )
+        return start, end
+
+    def _make_short_party_name(
+        self, name_part: str, parties: list[TexasCaseParty]
+    ) -> str:
+        """
+        Creates a shortened version of the party name based on the given list of parties.
+
+        Sometimes multiple parties to a case are listed in the same entry on TAMES so this method
+
+        1. Chooses the shortest party name between the list of parties and the long case name (these are often different)
+        2. Attempts to find and return the first party name in a string by splitting on semicolons. Some cases also list multiple parties in one entry separated by commas, but this is significantly more difficult to handle due to cases such as "lastname, firstname" and "company, LLC".
+
+        :param name_part: The initial part of the case name to process.
+        :param parties: A list of parties from which to determine the shortened name.
+
+        :return: The shortened party name derived from the input parameters.
+        """
+        if len(parties) == 0:
+            return name_part
+        if len(parties) == 1:
+            if len(parties[0]["name"]) < len(name_part):
+                return parties[0]["name"]
+            else:
+                return name_part
```

Comment:
I think we can use early returns to remove the else statements here and in the rest of this method where possible.